### PR TITLE
[move-format] Updated struct formatting

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR updates struct formatting (based on updated parser) to include new features: `public` keyword, positional fields, and postfix ability specification

## Test Plan 

Test has been updated to check the new features.

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts [hunk 1]
```diff
@@ -71,11 +72,13 @@ export function print(path: AstPath, options: ParserOptions, print: printFn) {
             //     f2: u64,
             // }
             return [
+                node.child(0).type == 'public' ? 'public ' : '',
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts:75]: nit: I believe `===` is stylistically preferred in TS code. I.e., if you _can_ use `===`, do. I searched our sui repo and we seem to be doing that for the most part, there are some `==` here and there

File: external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts [hunk 2]
```diff
@@ -129,7 +132,27 @@ export function print(path: AstPath, options: ParserOptions, print: printFn) {
...
     }
 }
+
+function breakable_comma_separated_list(path: AstPath,
+                                        node: SyntaxNode,
+                                        start: string,
+                                        end: string,
+                                        print: printFn) {
+
+    const items = Symbol('items');
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts:148]: formatting here seems a bit wacky, is this with prettier? but anyway 🤷

File: external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts [hunk 3]
```diff
@@ -129,7 +132,27 @@ export function print(path: AstPath, options: ParserOptions, print: printFn) {
...
+    return [
+        `${start}`,
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts:150]:
```suggestion
        start,
```

just that should work? same for `end`?

File: external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts [hunk 4]
```diff
@@ -112,7 +112,10 @@ export function print(path: AstPath, options: ParserOptions, print: printFn) {
                 node.namedChildren.length > 1 ? ': ' : '' ,
                 join(' + ', abilities),
             ];
-        case 'struct_def_fields':
+
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts:115]: newline needed?

**Review by rvantonder — APPROVED** [review]
lgtm!

File: external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts (hunk 3 above)
Comment (awelc) [reply external-crates/move/crates/move-analyzer/prettier-plugin/src/printer.ts:150]: Yeah, that was dumb :-) Initially I had a different implementation (of a different function that I subsequently re-wrote) where these included spaces before/after but at this point it just looks silly...

**Pull request merged**